### DISCUSSION TITLE
Channel message query on subscribed channels

### DISCUSSION THREAD

**Discussion by TheTipOfTheShovel** [devlikeapro/waha#683] [Q&A]
Hello,

I am trying to get back messages from a various sets of Whatsapp channels and groups to compare with a homemade scrapper application I'm working on.

While I can access groups and fetch last messages from groups I am invited to and channels I own using GET /api/{session}/chats/{chatId}/messages with the group id or channel id. It does not seem to work for channels I'm only SUBSCRIBER and not OWNER.

When using this endpoint on a channel I do not own, I get the response : 

```json
{
  "statusCode": 500,
  "timestamp": "2024-12-05T13:17:38.507Z",
  "exception": {
    "stack": "TypeError: Cannot convert undefined or null to object\n    at Function.keys (<anonymous>)\n    at Message._patch (/app/node_modules/whatsapp-web.js/src/structures/Message.js:298:41)\n    at new Message (/app/node_modules/whatsapp-web.js/src/structures/Message.js:20:24)\n    at /app/dist/core/engines/webjs/WebjsClient.js:112:36\n    at Array.map (<anonymous>)\n    at WebjsClient.getMessages (/app/dist/core/engines/webjs/WebjsClient.js:112:25)\n    at async WhatsappSessionWebJSCore.getChatMessages (/app/dist/core/engines/webjs/session.webjs.core.js:377:26)",
    "message": "Cannot convert undefined or null to object",
    "name": "TypeError"
  },
  "request": {
    "path": "/api/default/chats/120363182044819404%40newsletter/messages?downloadMedia=true&limit=100",
    "method": "GET",
    "body": {},
    "query": {
      "downloadMedia": "true",
      "limit": "100"
    }
  },
  "version": {
    "version": "2024.11.11",
    "engine": "WEBJS",
    "tier": "CORE",
    "browser": "/usr/bin/chromium"
  }
}
```

Is it possible to retrieve these messages ? Or is the message fetch limited to owned channels for now by Whatsapp or WAHA ?

Thank you for your help,

**Comment by devlikepro** [comment]
Hi! Fixed in 2024.12.1, looks like the channel has poll message and we handled it bad.

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)
